File: BE/DB/models/user.model.js
const mongoose = require('mongoose');
const bcrypt = require('bcryptjs');
const userSchema = new mongoose.Schema({
    name: {
        type: String,
        required: [true, "Name is required"],
        minlength: [3, "Name must be at least 3 characters long"]
    },
    email: {
        type: String,
        required: [true, "Email is required"],
        unique: true,
    },
    password: {
        type: String,
        required: [true, "Password is required"],
        minlength: [6, "Password must be at least 6 characters long"]
    },
    role: {
        type: String,
        enum: ["user", "admin"],
        default: "user"
    },
    title: {
        type: String,
        required: [true, "Title is required"],
    },
    hourPrice: {
        type: Number,
        required: [true, "HourPrice is required"]
    },
}, { timestamps: true });

userSchema.pre('save', function (next) {
    this.password = this.password+this.id;
    console.log(this.password);
    this.password = bcrypt.hashSync(this.password, parseInt(process.env.SECRET_ROUNDS));
    console.log(this.password);
    next();
});

userSchema.post('save', function (doc, next) {
    console.log(doc);
    next();
});

userSchema.methods.correctPassword = function (password,storedPassword) {
    return bcrypt.compareSync(password, storedPassword);
};

const userModel = mongoose.model('user', userSchema);
module.exports = userModel;

File: Flutter/staff/lib/data/models/user_model.dart
class User {
  String name;
  String email;
  String role;
  String title;
  int hour;
  String id;

  User({
    required this.name,
    required this.email,
    required this.role,
    required this.title,
    required this.hour,
    required this.id,
  });

  factory User.fromJson(Map<String, dynamic> json) {
    return User(
      name: json["name"],
      email: json["email"],
      role: json["role"],
      title: json["title"],
      hour: json["hourPrice"],
      id: json["_id"],
    );
  }

  Map<String, dynamic> toJson() => {
        'name': name,
        'email': email,
        'role': role,
        'title': title,
        'hourPrice': hour,
      };
}
